tests/test_image_driver.py

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python 
+''' VIC Image Driver testing '''
+
+import xarray as xr
+import numpy as np 
+import numpy.testing as npt 
+
+def assert_nan_equal(ds_domain, ds_output): 
+    """ 
+    test to see if nans in two data arrays are the same in order to check domain 
+    first dataarray is domain file, second dataarray is a variable in the output
+    dataset for image driver 
+    """
+
+    # get lat/lng coordinate names from domain file mask 
+    match_vars = ds_domain['mask'].coords
+    
+    # check to be sure that mask, lats and lons match between domain file and output file
+    
+    for var in match_vars: 
+        # raise AssertionError if they don't match
+        npt.assert_allclose(ds_output[var], ds_domain[var], equal_nan=True) 
+
+    # check that nans are occurring in the same place in the arrays 
+    
+    # check all variables in the dataset 
+    for da in ds_output.data_vars: 
+        
+        if (len(ds_output[da].dims) > 3):
+            # check all layers and timesteps 
+            da_isnull_reduced = ds_output[da].isnull().all(dim=('time', 'nlayer')) 
+            npt.assert_array_equal(da_isnull_reduced.values, 
+                                    np.isnan(ds_domain['mask']))
+
+        else: 
+            # check all timesteps 
+            da_isnull_reduced = ds_output[da].isnull().all(dim=('time')) 
+            npt.assert_array_equal(da_isnull_reduced.values, 
+                                    np.isnan(ds_domain['mask'])) 
```

Comment:
We shouldn't need to specify the `nlayer` dim option or even check if the variable is 4d.

What you want to do is take the difference in the dims between what the `mask` variable has and what your data variable has.  For example:

```Python
In [1]: ds_domain.mask.dims
Out[1]: ('x', 'y')
In [2]: ds_output[da].dims
Out[2]: ('x', 'y', 'time', 'snow_band', 'nlayer')
```

You can do this by taking the symmetric difference:

```Python
In [3]: a = ('x', 'y')
In [4]: b = ('x', 'y', 'time', 'snow_band', 'nlayer')
In [5]: set(a) ^ set(b)
Out[5]: {'nlayer', 'snow_band', 'time'}
```

Then you want to do you `...all(dim=(all_diff_dims)`.  Basically you are collapsing every data variable down to the shape of the mask.